File: Assets/ProjectGame/Lib/data/class_data_id/GuestCharacter/GuestCharacterTable.js
import { BaseClassDataID } from '../BaseClassDataID.js';
import { GuestCharacterRow } from './GuestCharacterRow.js';
import { GuestCharacterTableID } from './GuestCharacterTableID.js';

export class GuestCharacterTable extends BaseClassDataID {
    static Table = new Map();

    static _getEnum(name) {
        return GuestCharacterTableID[name];
    }

    static _getRowClass() {
        return GuestCharacterRow;
    }

    read(reader) {
        this.constructor.Table.clear();
        const rowCount = reader.readInt32();
        const colCount = reader.readInt32();
        for (let i = 0; i < colCount; i++) {
            const lenName = reader.readInt32();
            reader.readString(); // col name（ヘッダー読み飛ばし）
            const lenType = reader.readInt32();
            reader.readString(); // col type
        }
        for (let r = 0; r < rowCount; r++) {
            const enumInt = reader.readInt32();
            const enumVal = GuestCharacterTableID[Object.keys(GuestCharacterTableID).find(key => GuestCharacterTableID[key] === enumInt)] || enumInt;
            const row = new GuestCharacterRow();
            row.read(reader);
            this.constructor.Table.set(enumVal, row);
        }
    }

    static getRow(id) {
        return this.Table.get(id) ?? null;
    }
}

File: Assets/ProjectGame/Lib/data/class_data_id/GuestCharacter/GuestCharacterTable.py
from ..BaseClassDataID import BaseClassDataID
from .GuestCharacterRow import GuestCharacterRow
from .GuestCharacterTableID import GuestCharacterTableID
from typing import Dict

class GuestCharacterTable(BaseClassDataID):
    Table: Dict[GuestCharacterTableID, GuestCharacterRow] = {}

    @classmethod
    def _get_enum(cls, name: str):
        return GuestCharacterTableID[name]
    @classmethod
    def _get_row_class(cls):
        return GuestCharacterRow

    def read(self, reader):
        self.Table.clear()
        row_count = reader.read_int32()
        col_count = reader.read_int32()
        for _ in range(col_count):
            len_name = reader.read_int32()
            _ = reader.read_string()  # col name（ヘッダー読み飛ばし）
            len_type = reader.read_int32()
            _ = reader.read_string()  # col type
        for _ in range(row_count):
            enum_int = reader.read_int32()
            enum_val = GuestCharacterTableID(enum_int)
            row = GuestCharacterRow()
            row.read(reader)
            self.Table[enum_val] = row
